src/test/webapp/app/usermanagement/research-group/research-group-members/research-group-members.component.spec.ts

```diff
@@ -0,0 +1,228 @@
+import { ResearchGroupResourceApiService } from 'app/generated/api/researchGroupResourceApi.service';
+import { AccountService } from 'app/core/auth/account.service';
+import { UserShortDTO } from 'app/generated/model/userShortDTO';
+import { PageResponseDTOUserShortDTO } from 'app/generated/model/pageResponseDTOUserShortDTO';
+import { provideTranslateMock } from 'util/translate.mock';
+import { provideToastServiceMock, createToastServiceMock } from 'util/toast-service.mock';
+
+describe('ResearchGroupMembersComponent', () => {
+  let component: ResearchGroupMembersComponent;
+  let fixture: ComponentFixture<ResearchGroupMembersComponent>;
+  let mockResearchGroupService: {
+    getResearchGroupMembers: ReturnType<typeof vi.fn>;
+    removeMemberFromResearchGroup: ReturnType<typeof vi.fn>;
+  };
+  let mockToastService: ReturnType<typeof createToastServiceMock>;
+
+  const mockUserData: UserShortDTO = {
+    userId: 'user-1',
+    firstName: 'John',
+    lastName: 'Doe',
+    roles: [UserShortDTO.RolesEnum.Professor],
+  };
+
+  const mockCurrentUser: UserShortDTO = {
+    userId: 'current-user',
+    firstName: 'Current',
+    lastName: 'User',
+    roles: [UserShortDTO.RolesEnum.Admin],
+  };
+
+  const mockPageResponse: PageResponseDTOUserShortDTO = {
+    content: [mockUserData, mockCurrentUser],
+    totalElements: 2,
+  };
+
+  beforeEach(async () => {
+    mockResearchGroupService = {
+      getResearchGroupMembers: vi.fn(),
+      removeMemberFromResearchGroup: vi.fn(),
+    };
+
+    mockToastService = createToastServiceMock();
+
+    await TestBed.configureTestingModule({
+      imports: [ResearchGroupMembersComponent],
+      providers: [
+        { provide: ResearchGroupResourceApiService, useValue: mockResearchGroupService },
+        { provide: AccountService, useValue: { userId: 'current-user' } },
+        provideToastServiceMock(mockToastService),
+        provideTranslateMock(),
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(ResearchGroupMembersComponent);
+    component = fixture.componentInstance;
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should initialize with default values', () => {
+    expect(component.members()).toEqual([]);
+    expect(component.pageNumber()).toBe(0);
+    expect(component.pageSize()).toBe(10);
+    expect(component.total()).toBe(0);
+  });
+
+  it('should have correct columns configuration', () => {
+    fixture.detectChanges();
+
+    const columns = component.columns();
+    expect(columns).toHaveLength(4);
+    expect(columns[0].field).toBe('name');
+    expect(columns[0].header).toBe('researchGroup.members.tableColumns.name');
+    expect(columns[1].field).toBe('email');
+    expect(columns[2].field).toBe('role');
+    expect(columns[3].field).toBe('actions');
```

Comment:
I think if you already check the header for the first field, you can also check the other ones.

```suggestion
    const expected = [
      { field: 'name', header: 'researchGroup.members.tableColumns.name' },
      { field: 'email', header: 'researchGroup.members.tableColumns.email' },
      { field: 'role', header: 'researchGroup.members.tableColumns.role' },
      { field: 'actions', header: '' },
    ];
    expect(columns.map(c => ({ field: c.field, header: c.header }))).toEqual(expected);
```

Reply:
Maybe also adding another test or including it in this one, wether the templates were assigned to the correct columns.

```
it('should assign templates to correct columns', () => {
  fixture.detectChanges();
  
  const columns = component.columns();
  const nameColumn = columns.find(c => c.field === 'name');
  const actionsColumn = columns.find(c => c.field === 'actions');
  
  expect(nameColumn?.template).toBeTruthy();
  expect(actionsColumn?.template).toBeTruthy();
  expect(columns.find(c => c.field === 'email')?.template).toBeUndefined();
  expect(columns.find(c => c.field === 'role')?.template).toBeUndefined();
});
```